alma/includes/AlmaEligibilityHelper.php

```diff
@@ -66,48 +66,74 @@ private static function checkPnXBounds($cart)
 
     public static function eligibilityCheck($context)
     {
-        $pnxBounds = self::checkPnXBounds($context->cart);
-        // If we got an array, then the cart is not eligible because not within the returned bounds
-        if (is_array($pnxBounds)) {
-            // Mock Alma's Eligibility object
-            $eligibility = new stdClass();
-            $eligibility->isEligible = false;
-            $eligibility->constraints = array(
-                "purchase_amount" => array(
-                    "minimum" => $pnxBounds[0],
-                    "maximum" => $pnxBounds[1]
-                )
-            );
-
-            return $eligibility;
-        }
-
-        $paymentData = PaymentData::dataFromCart($context->cart, $context);
-        if (!$paymentData) {
-            AlmaLogger::instance()->error('Cannot check cart eligibility: no data extracted from cart');
-
-            return null;
-        }
-
+        $eligibilities = array();
+        $activePlans = array();
+        $purchaseAmount = (float) Tools::ps_round((float)$context->cart->getOrderTotal(true, Cart::BOTH), 2) * 100;
         $alma = AlmaClient::defaultInstance();
         if (!$alma) {
             AlmaLogger::instance()->error('Cannot check cart eligibility: no API client');
+            return array();
+        }
+        if(0 === AlmaSettings::installmentPlansMaxN()){
+            return array();
+        }
+        $n = 1;
+        while ($n < AlmaSettings::installmentPlansMaxN()) {
+            ++$n;
+            if (!AlmaSettings::isInstallmentPlanEnabled($n)) {
+                continue;
+            } else {
```

Comment:
Add this to `AlmaSettings`:

```php
    public static function activeInstallmentsCounts()
    {
        $installmentsCounts = array();

        for ($n = 2; $n <= self::installmentPlansMaxN(); $n++) {
            if (self::isInstallmentPlanEnabled($n)) {
                $installmentsCounts[] = $n;
            }
        }

        return $installmentsCounts;
    }
```

and use `foreach(AlmaSettings::activeInstallmentsCounts() as $n)` :)